ledger-bridge.js

```diff
@@ -34,40 +49,82 @@ export default class LedgerBridge {
         window.parent.postMessage(msg, '*')
     }
 
+    delay (ms) {
+        return new Promise((success) => setTimeout(success, ms))
+    }
+
+    checkTransportLoop (i) {
+        const iterator = i || 0
+        return WebSocketTransport.check(BRIDGE_URL).catch(async () => {
```

Comment:
So this only works if `WebSocketTransport.check(BRIDGE_URL)` throws? I'm not sure I follow what's going on here.

Reply:
Oh wait, I have a guess. Is this to ensure the connection is only re-established when it has been broken? To avoid reconnecting with each interaction.

Reply:
No; this checks for a connection/unlock every second during any unlock attempt.  This, for example, gives the user time to unlock their Ledger and open the Ethereum app before completely bombing out.

Reply:
Sorry I meant specifically why is this block executed only when `WebSocketTransport.check(BRIDGE_URL)` throws. See that it's all in the `.catch` handler - there is no `.then` handler. If  `WebSocketTransport.check(BRIDGE_URL)` succeeds, `this.transport` and `this.app` don't get set.

Reply:
Oh dear, I see what happened. This comment was left on the wrong line, my bad.

I saw this same pattern below on line 71: `await WebSocketTransport.check(BRIDGE_URL).catch(...`

Then I must have looked away, looked back, and scanned for this same pattern before leaving a comment. I hadn't realized there were two lines like this :facepalm:

Reply:
So to recap:
>Is this to ensure the connection is only re-established when it has been broken? To avoid reconnecting with each interaction.

^ I meant to ask about line 71 with this question.

Reply:
Whew, I was confused.  Yes, that is correct.